src/node_buffer.h

```diff
@@ -47,6 +47,11 @@ NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                            char* data,
                                            size_t len);
 
+// public constructor - data is used if memory was allocated, copied otherwise.
+NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
+                                           MaybeStackBuffer<char>* buf,
+                                           size_t len);
```

Comment:
I think this would be easier to integrate in `MaybeStackBuffer` as a method. Imo it kind of makes sense there, and on the more technical side, `node_buffer.h` is a public header while `util.h` is an internal one.